packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.test.tsx

```diff
@@ -279,31 +277,57 @@ describe('routes/Product/SubscriptionCreate', () => {
     };
   }
 
+  enum CustomerType {
+    NoCustomer,
+    WithCustomer,
+    WithCustomerCard,
+    IapCustomerOnly,
+  }
+
   const commonPaymentSubmissionTest =
     ({
       // "Customer" here is one _without_ any subs or CC info.
       // It affects the number of calls to apiClientOverrides.apiCreateCustomer.
-      withCustomer,
-      withExistingCard = false,
+      customerType,
     }: {
-      withCustomer: boolean;
-      withExistingCard?: boolean;
+      customerType: CustomerType;
     }) =>
     async () => {
-      const customer = withExistingCard ? CUSTOMER : withCustomer ? {} : null;
-      const expectedCreateCustomerCalls = customer === null ? 1 : 0;
-      const expectedCreatePaymentMethodCalls = !withExistingCard ? 1 : 0;
+      let customer;
+      switch (customerType) {
+        case CustomerType.NoCustomer:
```

Comment:
A valid customer type is NoCustomer? 🤔

Reply:
Yup if a user doesn't have any subscriptions, or hasn't tried signing up for any subscriptions, they won't have a `customer`.

A `customer` in the front end will only exist if they have a Stripe Customer, or an IAP subscription.